source/docs/software/basic-programming/functions-as-data.rst

```diff
@@ -0,0 +1,71 @@
+Functions as Data
+=================
+
+Regardless of programming language, one of the first things anyone learns to do when programming a computer is to write a function (also known as a "method" or a "subroutine").  Functions are a fundamental part of organized code - writing functions lets us avoid duplicating the same piece of code over and over again.  Instead of writing duplicated sections of code, we call a single function that contains the code we want to execute from multiple places.  If the section of code needs some additional information about its surrounding context to run, we pass those to the function as parameters.
```

Comment:
I agree with the deduplication argument, but functions are also good for behavioral abstraction, even if they're used once. Rather than writing a giant function with comments above each block of behavior, you can move them into well-named functions so the high-level control flow is easier to discern.

Reply:
for sure, this isn't an exhaustive list of reasons why functions are good